src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -471,6 +472,16 @@ private IEnumerable<(ManifestId id, ManifestVersion version, SdkFeatureBand feat
                     ManifestVersion manifestVersion;
                     SdkFeatureBand manifestFeatureBand;
                     var parts = manifest.Value.Split('/');
+                    
+                    string manifestVersionString = (parts[0]).ToString();
+                    if (!FXVersion.TryParse(manifestVersionString, out FXVersion _version))
+                    {
+                        if (string.IsNullOrWhiteSpace(manifestVersionString))
+                        {
+                            throw new ArgumentNullException(nameof(manifestVersionString));
+                        }
+                        throw new FormatException(String.Format(LocalizableStrings.InvalidVersionForWorkload, manifest.Key, manifestVersionString));
+                    }
                     manifestVersion = new ManifestVersion(parts[0]);
```

Comment:
The `ManifestVersion` constructor should be fixed too.  We can also check in this method as we can generate a better error message which incudes the manifest ID.

It's probably a good idea to add a constructor overload to `ManifestVersion` which takes an `FXVersion` as a parameter so we don't have to parse it twice.